File: Minimum Time Visiting All Points 1266/solution.py
class Solution(object):
    def minTimeToVisitAllPoints(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        second_times = 0

        first_point = points.pop(0)

        #遍历
        for point in points:
            # first coordinate x
            first_x = first_point[0]

            # first coordinate y
            first_y = first_point[1]

            # next coordinate x
            next_x = point[0]

            # next coordinate y
            next_y = point[1]

            while (True):
                #move target
                if ( first_x == next_x and first_y == next_y ):
                    break

                if ( first_x == next_x and first_y != next_y ):
                    #move vertically
                    if ( next_y > first_y ):
                        first_y = first_y + 1
                    else:
                        first_y = first_y - 1
                elif ( first_x != next_x and first_y == next_y ):
                    #move horizontally
                    if ( next_x > first_x ):
                        first_x = first_x + 1
                    else:
                        first_x = first_x - 1
                else:
                    #move diaonally
                    if ( next_x > first_x ):
                        first_x = first_x + 1
                    else:
                        first_x = first_x - 1

                    if ( next_y > first_y ):
                        first_y = first_y + 1
                    else:
                        first_y = first_y - 1

                second_times = second_times + 1


            first_point = point

        return second_times

        

File: Minimum Time Visiting All Points 1266/solution.cpp
class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        
        int second_times = 0;
        vector<vector<int>>::iterator it_point_first = points.begin();
        for ( vector<vector<int>>::iterator it_point_next = it_point_first+1; it_point_next != points.end(); it_point_next++ )
        {
            //first coordinate x
            int first_x = (*it_point_first).at(0);
            
            //first coordinate y
            int first_y = (*it_point_first).at(1);
            
            //next coordinate x
            int next_x = (*it_point_next).at(0);
            
            //next coordinate y
            int next_y = (*it_point_next).at(1);
            
            while( true )
            {
                //come to target
                if ( first_x == next_x && first_y == next_y )
                    break;
                
                //move vertically
                if ( first_x == next_x && first_y != next_y )
                {
                    if ( next_y > first_y ) first_y++;
                    else    first_y--;
                }
                else if ( first_y == next_y && first_x != next_x )
                {
                    //move horizontally
                    if ( next_x > first_x ) first_x++;
                    else    first_x--;
                }
                else
                {
                    //move diaonally
                    if ( next_x > first_x ) first_x++;
                    else first_x--;
                    
                    if ( next_y > first_y ) first_y++;
                    else first_y--;
                }
                
                second_times++;
                
            }
            
            //swap point
            it_point_first = it_point_next;
            
            
        }
        
        return second_times;
    }
};